source/fundamentals/crud/write-operations/change-a-document.txt

```diff
@@ -0,0 +1,202 @@
+document:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 40 60
+
+   * - Update Operator
+     - Description
+
+   * - ``$set``
+     - replaces the value of a field with a specified value or creates a new
+       field with a specified value
+
+   * - ``$inc``
+     - increments or decrements field values
+
+   * - ``$rename``
+     - renames fields
+
+   * - ``$unset``
+     - removes fields
+     
+   * - ``$mul``
+     - multiplies the value of a field by a specified number
+
+See the MongoDB server manual for a :manual:`complete list of update operators
+and descriptions </reference/operator/update-field/>`.
+
+.. note:: Aggregation Pipelines in Update Operations
+
+   If you are using MongoDB Version 4.2 or later, you can use aggregation
+   pipelines made up of a subset of aggregation stages in update operations. For
+   more information on the aggregation stages MongoDB supports in
+   aggregation pipelines, see our tutorial on performing
+   :manual:`updates with aggregation pipelines </tutorial/update-documents-with-aggregation-pipeline/>`.
+
+Example
+~~~~~~~
+
+The following document describes an employee:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+      "_id" : 2158,
+      "name" : "Mary Shelley",
+      "department" : "Marketing",
+      "role" : "Marketing Analyst",
+      "bonus" : 2500,
+      ...
+   }
+
+The following code shows how you can use the ``$set`` update operator to
+change the value of the ``role`` field to "Marketing Director" and the
+``$inc`` update operator to increment the value of the ``bonus`` field
+by 2000:
+
+.. code-block:: go
+
+   filter := bson.D{{"_id", 2158}}
+   update := bson.D{{"$set", bson.D{{"role", "Marketing Director"}}}, {"$inc", bson.D{{"bonus", 2000}}}}
+
+   result, err := collection.UpdateOne(context.TODO(), filter, update)
+
+The updated document should resemble the following, with updated values in
+the ``role`` and ``bonus`` fields and all other values unchanged:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+      "_id" : 2158,
+      "name" : "Mary Shelley",
+      "department" : "Marketing",
+      "role" : "Marketing Director",
+      "bonus" : 4500,
+      ...
+   }
+
+If the query filter fails to match any documents in a collection, the
+update operation doesn't make any changes. You can configure an update
+operation to perform an :doc:`upsert </fundamentals/crud/write-operations/upsert>`, 
+in which the operation attempts to perform an update but inserts
+a new document if no documents are matched:
+
+.. code-block:: go
+
+   opts := options.Update().SetUpsert(true)
+   ...
+   result, err := collection.UpdateOne(context.TODO(), filter, update, opts)
+
+You can't modify the ``_id`` field of a document nor change a field to
+a value that violates a unique index constraint. For more information
+about unique index constraints, see the MongoDB server manual entry on :manual:`unique indexes </core/index-unique/>`.
```

Comment:
```suggestion
If zero documents match the query filter in the update operation, updateOne() makes no changes to the preceding document. See our <TODO: link> upsert guide to learn how to insert a new document instead of updating one if no documents match.
```

Suggestion: I think this section can be simplified since this information will be duplicated on the upsert page.

Reply:
accepted